Client/Frontend/Browser/FaviconManager.swift

```diff
@@ -54,10 +54,10 @@ class FaviconManager: TabHelper {
         return all(deferreds.map({$0()}))
     }
     
-    func getFavicon(_ tab: Tab, iconUrl: URL, currentURL: URL, icon: Favicon, profile: Profile) -> Deferred<Maybe<Favicon>> {
+    func getFavicon(_ tab: Weak<Tab>, iconUrl: URL, currentURL: URL, icon: Favicon, profile: Profile) -> Deferred<Maybe<Favicon>> {
         let deferred = Deferred<Maybe<Favicon>>()
         let manager = SDWebImageManager.shared()
-        let options: [SDWebImageOptions] = tab.isPrivate ? [.lowPriority, .cacheMemoryOnly] : [.lowPriority]
+        let options: [SDWebImageOptions] = tab.object?.isPrivate ?? false ? [.lowPriority, .cacheMemoryOnly] : [.lowPriority]
         let url = currentURL.absoluteString
         let site = Site(url: url, title: "")
 
```

Comment:
I think you could just declare a `weak var weaktab = tab` here and use that inside the block. That should fix the issue?

Reply:
Tab has strongly been captured by this point, I'm not clear on your suggestion.

Reply:
Maybe @rnewman has come across this (pondered the enforced weak ownership of Tab) as he was talking about a more abstract interface to Tab list that would avoid shared ownership. I think that would be a bigger change though.

Reply:
The reason why the tab is being retained is because of the nested function (`loadImageCompleted`). The function is retaining a reference to `tab`. Instead of using tab directly inside this nested function we use a weak one. Tab will no longer be retained. 

I think there are a few ways you could fix this
1. keep `loadImageCompleted` where it is but create a weak ref to tab so it does not up the retain count
2. dont make `loadImageCompleted` a nested function.

Reply:
Kinda like this. It should work. 
```
weak var tabweak = tab
func loadImageCompleted(_ img: UIImage?, _ url: URL?) {

    guard let img = img, let myTab = tabweak, let urlString = url?.absoluteString else {
        deferred.fill(Maybe(failure: FaviconError()))
        return
    }

    let fav = Favicon(url: urlString, date: Date(), type: icon.type)
    fav.width = Int(img.size.width)
    fav.height = Int(img.size.height)

    if !myTab.isPrivate {....
```

Reply:
Cool I'll try something like that, I thought this looked more minimal in terms of changes, but now I see the patch isn't minimal compared to making funcs into closures with weak captures.